sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/options.py

```diff
@@ -0,0 +1,175 @@
+        storage_maintenance_period: Local storage maintenance interval in seconds.
+        storage_max_size: Local storage maximum size in bytes.
+        storage_path: Local storage file path.
+        storage_retention_period: Local storage retention period in seconds
+        timeout: Request timeout in seconds
+    """
+
+    __slots__ = (
+        "connection_string",
+        "endpoint",
+        "instrumentation_key",
+        "proxies",
+        "storage_maintenance_period",
+        "storage_max_size",
+        "storage_path",
+        "storage_retention_period",
+        "timeout",
+    )
+
+    def __init__(
+        self,
+        connection_string: str = None,
+        instrumentation_key: str = None,
+        proxies: typing.Dict[str, str] = None,
+        storage_maintenance_period: int = 60,
+        storage_max_size: int = 50 * 1024 * 1024,
+        storage_path: str = None,
+        storage_retention_period: int = 7 * 24 * 60 * 60,
+        timeout: int = 10.0,  # networking timeout in seconds
+    ) -> None:
+        self.connection_string = connection_string
+        self.instrumentation_key = instrumentation_key
+        self.proxies = proxies
+        self.storage_maintenance_period = storage_maintenance_period
+        self.storage_max_size = storage_max_size
+        self.storage_path = storage_path
+        self.storage_retention_period = storage_retention_period
+        self.timeout = timeout
+        self.endpoint = ""
+        self._initialize()
+        self._validate_instrumentation_key()
+
+    def _initialize(self) -> None:
+        # connection string and ikey
+        code_cs = parse_connection_string(self.connection_string)
+        code_ikey = self.instrumentation_key
+        env_cs = parse_connection_string(
+            os.getenv("APPLICATIONINSIGHTS_CONNECTION_STRING")
+        )
+        env_ikey = os.getenv("APPINSIGHTS_INSTRUMENTATIONKEY")
+
+        # The priority of which value takes on the instrumentation key is:
+        # 1. Key from explicitly passed in connection string
+        # 2. Key from explicitly passed in instrumentation key
+        # 3. Key from connection string in environment variable
+        # 4. Key from instrumentation key in environment variable
+        self.instrumentation_key = (
+            code_cs.get(INSTRUMENTATION_KEY)
+            or code_ikey
+            or env_cs.get(INSTRUMENTATION_KEY)
+            or env_ikey
+        )
+        # The priority of the ingestion endpoint is as follows:
+        # 1. The endpoint explicitly passed in connection string
+        # 2. The endpoint from the connection string in environment variable
+        # 3. The default breeze endpoint
+        self.endpoint = (
+            code_cs.get(INGESTION_ENDPOINT)
+            or env_cs.get(INGESTION_ENDPOINT)
+            or "https://dc.services.visualstudio.com"
+        )
+
+        # proxies
+        if self.proxies is None:
```

Comment:
Do we need this default now that we are using `ProxyPolicy`?

Reply:
We use this as parameter to instantiate the ProxyPolicy, there are suggestions on changing configuration of the exporter and allowing users to configure the policies directly, we can address this when that happen